install-deps.sh

```diff
@@ -146,19 +146,51 @@ function install_pkg_on_ubuntu {
     fi
 }
 
+boost_ver=1.82
+
+function clean_boost_on_ubuntu {
+    ci_debug "Running clean_boost_on_ubuntu() in install-deps.sh"
+    # Find currently installed version. This may not work if there are
+    # multiple versions installed.
+    local installed_ver=$(apt -qq list --installed ceph-libboost*-dev 2>/dev/null |
+                              grep -e 'libboost[0-9].[0-9]\+-dev' |
+                              cut -d' ' -f2 |
+                              cut -d'.' -f1,2)
+    # If installed_ver contains a space, we can't really count on it,
+    # but otherwise, bail out if the version installed is the version
+    # we want.
+    if test -n "$installed_ver" && echo "$installed_ver" | grep -v -q ' '; then
+        if echo "$installed_ver" | grep -q "^$ver"; then
+            return
+        fi
+    fi
+
+    # Historical packages
+    $SUDO rm -f "/etc/apt/sources.list.d/ceph-libboost*.list"
+    # Currently used
+    $SUDO rm -f "/etc/apt/sources.list.d/libboost.list"
+    # Refresh package list so things aren't in the available list.
+    $SUDO apt-get update -y
+    # Remove all ceph-libboost dev packages except, possibly, those of
+    # the version we currently want.
+    apt -qq list --installed 'ceph-libboost*-dev' 2> /dev/null | \
+	grep -F -v "$boost_ver" | \
```

Comment:
Now that a check for the installed version has been added at the top, do we still need this filter here?  Control gets to this point if either a) no boost packages are installed, b) a single version but not the one desired is installed or c) multiple versions are installed.  Since we are removing all related repositories, it feels weird to potentially leave some packages installed from those repositories behind.

I would treat (c) as a "something got messed up around here" case and remove all related packages, letting `install_boost_on_ubuntu()` install the desired version from scratch.